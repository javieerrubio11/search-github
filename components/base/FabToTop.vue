<template>
  <v-fab-transition>
    <v-btn
      v-show="fab"
      v-scroll="onScroll"
      title="Scroll to top"
      bottom
      class="transition-swing mb-10"
      color="primary"
      fab
      fixed
      small
      right
      style="z-index: 6"
      @click="toTop"
    >
      <v-icon>mdi-chevron-up</v-icon>
    </v-btn>
  </v-fab-transition>
</template>

<script>
// Globals
export default {
  name: 'DefaultFabToTop',
  data: () => ({ fab: false }),
  methods: {
    onScroll() {
      if (!window) return

      const top = window.pageYOffset || document.documentElement.offsetTop || 0
      this.fab = top > 300
    },
    toTop() {
      if (this.$route.hash) {
        this.$router.push({ hash: '' })
      }
      this.$vuetify.goTo(0)
    }
  }
}
</script>
