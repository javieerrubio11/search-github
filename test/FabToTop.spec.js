import { mount, shallowMount } from '@vue/test-utils'
import FabToTop from '@/components/base/FabToTop.vue'

const factory = () => {
  return shallowMount(FabToTop, {
  });
};

describe('FabToTop', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(FabToTop)
    expect(wrapper.vm).toBeTruthy()
  })

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
})
