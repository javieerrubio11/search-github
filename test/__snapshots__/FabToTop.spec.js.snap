// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FabToTop renders properly 1`] = `
"<v-fab-transition>
  <v-btn title=\\"Scroll to top\\" bottom=\\"\\" color=\\"primary\\" fab=\\"\\" fixed=\\"\\" small=\\"\\" right=\\"\\" class=\\"transition-swing mb-10\\" style=\\"z-index: 6; display: none;\\">
    <v-icon>mdi-chevron-up</v-icon>
  </v-btn>
</v-fab-transition>"
`;
